testsuite/MDAnalysisTests/analysis/test_distances.py

```diff
@@ -106,43 +108,76 @@ def test_box_sparse(self, coord, box, shape, res_pbc):
         assert_equal(contacts.toarray(), res_pbc)
 
 
+
 class TestDist(object):
 
     @staticmethod
     @pytest.fixture()
     def ag():
         u = MDAnalysis.Universe(GRO)
-        return u.atoms[:20]
+        return u.atoms[1:10]
 
     # TODO: How are ag and ag2 different?!
     @staticmethod
     @pytest.fixture()
     def ag2():
         u2 = MDAnalysis.Universe(GRO)
-        return u2.atoms[:20]
+        return u2.atoms[11:20]
 
+    @staticmethod
+    @pytest.fixture()
+    def box():
+        return np.array([8, 8, 8, 90, 90, 90], dtype=np.float32)
+    
     @staticmethod
     @pytest.fixture()
     def expected(ag, ag2):
-        np.random.shuffle(ag2.positions)
+        
         return np.diag(scipy.spatial.distance.cdist(
             ag.positions, ag2.positions)
         )
 
+    @staticmethod
+    @pytest.fixture()
+    def expected_box(ag, ag2):
+
+        result= np.empty((0))
+        for i in range(len(ag)):
+            x= abs(ag.positions[i][0]-ag2.positions[i][0])
+            if x>4:
+                x=8-4
+            x=x*x
+            y= abs(ag.positions[i][1]-ag2.positions[i][1])
+            if y>4:
+                y=8-4
+            y=y*y
+            z= abs(ag.positions[i][2]-ag2.positions[i][2])
+            if z>4:
+                z=8-4
+            z=z*z
+            result=np.append(result,np.sqrt(x+y+z))
+
```

Comment:
You can substitute the loop for numpy array manipulation:
```
rp = np.abs(ag.positions - ag2.positions)
rp = np.where(rp > 4, 8.0 - rp, rp)
return np.sqrt(np.square(rp).sum(axis=1))
```

Reply:
yes sure, I’ll remember to write neater code from next time.